synapse/util/async_helpers.py

```diff
@@ -796,7 +800,13 @@ def timeout_deferred(
     Args:
         deferred: The Deferred to potentially timeout.
         timeout: Timeout in seconds
-        reactor: The twisted reactor to use
+        cancel_on_shutdown: Whether this call should be tracked for cleanup during
+                shutdown. Any call with a long delay, or that is created infrequently,
+                should be tracked. Calls which are short or of 0 delay don't require
+                tracking since the small delay after shutdown before they trigger is
+                immaterial. It's not worth the overhead to track those calls as it blows up
+                the tracking collection on large server instances.
```

Comment:
```suggestion
        cancel_on_shutdown: Whether this call should be tracked for cleanup during
            shutdown. Any call with a long delay, or that is created infrequently,
            should be tracked. Calls which are short or of 0 delay don't require
            tracking since the small delay after shutdown before they trigger is
            immaterial. It's not worth the overhead to track those calls as it blows up
            the tracking collection on large server instances.
```

(nested too many levels)